### PR TITLE
Clarify that `Location.assign` and `Location.replace` also accept `URL`s

### PR DESCRIPTION
**Pull request by silverwind**
### Description

`Location.assign` and `Location.replace` also accept `URL`s

### Motivation

This works in all browsers I tested (Firefox, Chrome, Safari, PaleMoon), so MDN was incorrect.

### Additional details

TypeScript has it correct already:

https://github.com/microsoft/TypeScript/blob/e6ef279403d86440600c866d53839a3e695220d3/src/lib/dom.generated.d.ts#L15107
https://github.com/microsoft/TypeScript/blob/e6ef279403d86440600c866d53839a3e695220d3/src/lib/dom.generated.d.ts#L15119

### CODE REVIEW

**Comment by Josh-Cena** [comment]
There's nothing special about passing URLs to these APIs. They merely coerce everything to strings. Check the spec: https://html.spec.whatwg.org/multipage/nav-history-apis.html#the-location-interface `undefined assign(USVString url);`

The reason TypeScript needs to declare the type like this is because it doesn't have a type that means "all objects with sensible `toString` behavior" and presumably someone complained that they can't pass a URL directly without converting it to string first. But as far as the IDL and the underlying implementation is concerned, `URL` is not a "blessed type" as an input.

**Comment by silverwind** [comment]
Saying it accepts "a string" is still incorrect, thought. Maybe it should say "a string, or any object with a `.toString()` method"?

**Comment by sideshowbarker** [comment]
> I think we have precedents of doing similar stuff elsewhere, so you may want to search for one, but the idea makes sense to me.

Yeah, for one relevant precedent, see the wording about the `url` param in the https://developer.mozilla.org/en-US/docs/Web/API/URL/URL#url section of the “URL: URL() constructor” article:

> _A string or any other object with a [stringifier](https://developer.mozilla.org/en-US/docs/Glossary/Stringifier) that represents an absolute URL or a relative reference to a base URL._

And in that same article, see the note at https://developer.mozilla.org/en-US/docs/Web/API/URL/URL#sect2:

> _The `url` and `base` arguments will each be stringified from whatever value you pass, such as an [HTMLAnchorElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLAnchorElement) or [HTMLAreaElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLAreaElement) element, just like with other Web APIs that accept a string. In particular, you can use an existing [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL) object for either argument, and it will be stringified from the object's [href](https://developer.mozilla.org/en-US/docs/Web/API/URL/href) property._

**Comment by silverwind** [comment]
> > I think we have precedents of doing similar stuff elsewhere, so you may want to search for one, but the idea makes sense to me.
> 
> Yeah, for one relevant precedent, see the wording about the `url` param in the https://developer.mozilla.org/en-US/docs/Web/API/URL/URL#url section of the “URL: URL() constructor” article:
> 
> > _A string or any other object with a [stringifier](https://developer.mozilla.org/en-US/docs/Glossary/Stringifier) that represents an absolute URL or a relative reference to a base URL._
> 
> And in that same article, see the note at https://developer.mozilla.org/en-US/docs/Web/API/URL/URL#sect2:
> 
> > _The `url` and `base` arguments will each be stringified from whatever value you pass, such as an [HTMLAnchorElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLAnchorElement) or [HTMLAreaElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLAreaElement) element, just like with other Web APIs that accept a string. In particular, you can use an existing [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL) object for either argument, and it will be stringified from the object's [href](https://developer.mozilla.org/en-US/docs/Web/API/URL/href) property._

I agree and have changed to use the form "'A string or any other object with a stringifier'" of that wording now.

File: files/en-us/web/api/location/assign/index.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ assign(url)
 ### Parameters
 
 - `url`
-  - : A string containing the URL of the page to navigate to; for example, an absolute URL such as `https://developer.mozilla.org/en-US/docs/Web/API/Location/reload`, or a relative URL — such as `/Web` (just a path, for navigating to another document at the same origin) or `#specifications` (just a fragment string, for navigating to some part of the same page), and so on.
+  - : A string or any other object with a {{Glossary("stringifier")}} containing the URL of the page to navigate to; for example, an absolute URL such as `https://developer.mozilla.org/en-US/docs/Web/API/Location/reload`, or a relative URL — such as `/Web` (just a path, for navigating to another document at the same origin) or `#specifications` (just a fragment string, for navigating to some part of the same page), and so on.
```
Comment (Josh-Cena) [inline files/en-us/web/api/location/assign/index.md:25]:
```suggestion
  - : A string or any other object with a {{Glossary("stringifier")}}, such as a {{domxref("URL")}} object, containing the URL of the page to navigate to; for example, an absolute URL such as `https://developer.mozilla.org/en-US/docs/Web/API/Location/reload`, or a relative URL — such as `/Web` (just a path, for navigating to another document at the same origin) or `#specifications` (just a fragment string, for navigating to some part of the same page), and so on.
```

File: files/en-us/web/api/location/replace/index.md [hunk 2]
```diff
@@ -24,7 +24,7 @@ replace(url)
 ### Parameters
 
 - `url`
-  - : A string containing the URL of the page to navigate to.
+  - : A string or any other object with a {{Glossary("stringifier")}} containing the URL of the page to navigate to.
```
Comment (Josh-Cena) [inline files/en-us/web/api/location/replace/index.md:27]:
```suggestion
  - : A string or any other object with a {{Glossary("stringifier")}}, such as a {{domxref("URL")}} object, containing the URL of the page to navigate to.
```

**Review by Josh-Cena — APPROVED** [review]

**Pull request merged**